src/Things/Structures/Structure.cpp

```diff
@@ -5,6 +5,10 @@
 
 #include "../../Constants.h"
 
+#include <random>
+
+std::random_device rd;
+std::mt19937 generator(rd());
```

Comment:
For purposes such as network sync or game replay saving, we should be careful about uncontrolled use of random number generators. It might be best to have a central place to store a random number generator for the whole game, and to consistently use the same generator.

Reply:
I was thinking the same thing. You really shouldn't have more than one RNG per thread and it certainly shouldn't be initalized more than once (or reinitialzied, if you store the seed in the save game file.)

Reply:
I was just looking for existing practice. The `Population` code also just creates a random number generator and uses it. As such, I would consider this an external concern, though we really should be opening an issue to track this, and coming up with a general solution so all game components have access to a random number generator.

Reply:
Added https://github.com/lairworks/nas2d-core/issues/453 to track.